configuration/src/main/java/com/synopsys/integration/configuration/help/PropertyConfigurationHelpContext.java

```diff
@@ -68,27 +73,40 @@ public PropertyConfigurationHelpContext(final PropertyConfiguration propertyConf
         this.propertyConfiguration = propertyConfiguration;
     }
 
-    public void printCurrentValues(Consumer<String> logger, List<Property> knownProperties, Map<String, String> additionalNotes) {
+    public void printCurrentValues(Consumer<String> logger, Set<Property> knownProperties, Map<String, String> additionalNotes) {
         logger.accept("");
         logger.accept("Current property values:");
         logger.accept("--property = value [notes]");
         logger.accept(StringUtils.repeat("-", 60));
 
-        PropertyInfoCollector propertyInfoCollector = new PropertyInfoCollector(propertyConfiguration);
-        List<PropertyInfo> propertyInfoList = propertyInfoCollector.collectPropertyInfo(knownProperties, PropertyInfoCollector.maskPasswordsAndTokensPredicate());
-        for (PropertyInfo propertyInfo: propertyInfoList) {
-            String sourceName = propertyConfiguration.getPropertySource(propertyInfo.getProperty()).orElse("unknown");
+        Map<String, String> sortedMaskedRawPropertyKeyValues = getSortedMaskedRawPropertyKeyValues(knownProperties);
+        for (Map.Entry<String, String> rawKeyValue: sortedMaskedRawPropertyKeyValues.entrySet()) {
+            String sourceName = propertyConfiguration.getPropertySource(rawKeyValue.getKey()).orElse("unknown");
             String sourceDisplayName = knownSourceDisplayNames.getOrDefault(sourceName, sourceName);
 
-            String notes = additionalNotes.getOrDefault(propertyInfo.getKey(), "");
+            String notes = additionalNotes.getOrDefault(rawKeyValue.getKey(), "");
 
-            logger.accept(propertyInfo.getKey() + " = " + propertyInfo.getValue() + " [" + sourceDisplayName + "] " + notes);
+            logger.accept(rawKeyValue.getKey() + " = " + rawKeyValue.getValue() + " [" + sourceDisplayName + "] " + notes);
         }
 
         logger.accept(StringUtils.repeat("-", 60));
         logger.accept("");
     }
 
+    private Map<String, String> getSortedMaskedRawPropertyKeyValues(Set<Property> knownProperties) {
```

Comment:
I think this can be simplified. You don't actually need a sorted map (and I find it strange we are trying to sort the map as maps are generally not sorted, similar to a Set imo). At some point prior you had a List of ordered properties; why not just iterate that and do an containsKey check on the map?

Reply:
I figure it's nice to have the properties sorted when they're displayed, helps if you're looking for one in particular.  I changed the parameter to take in a set, as I thought that would be more reflective of the argument (there shouldn't be duplicate properties, no?).  If that was misguided, then I can revert it back to a list and can take the approach you suggest

Reply:
Well you can sort the printed properties without sorting the map (sort the list, access the map via the sorted list). It might just be me but Map and Set I think should generally not be sorted - I don't actually know what the opinion is at large there. A list has a natural order and makes sense to be sorted. So sorting a list and using the list to access the map seems more reasonable to me. I'm fine with it staying a sorted map but wanted bring it up.

Reply:
Sure.  So I guess the question is should knownProperties be a list to make the sorting more appropriate?  Or should it be a set and we don't sort at all?  Or should it be a set and we sort the set, use that to query the map?  Or should it be a set and we use it to create a sorted map?  I think I lean with what I currently have.  I could make a poll?

Reply:
Well we should sort it. I think a poll between sorting the map or using a sorted list of keys would be interesting and I'd accept whatever the answer is.